### PR TITLE
#15 generate CSV output

### PR DESCRIPTION
**Pull request by chhabriv**
- Added FileWriter for CSV output
    - includes model -> df conversion
    - df -> CSV
- Wired sample call from main -- temp, will add test cases based on comments
- Added sample output CSV
- Organised imports using i[sort python](https://github.com/timothycrosley/isort)

image

### CODE REVIEW

**Comment by rupdeb** [comment]
why are there so many changes?

File: code/models/OutputDataModel.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+class OutputDataModel:
+    def __init__(self):
+        LOG.debug(f"{__name__} init")
+        self.coreMetrics = CoreMetrics()
+        self.readabilityMetrics = ReadabilityMetrics()
+        self.category = ""
+        self.description = ""
+
+    def setCoreMetrics(self, coreMetrics: CoreMetrics) -> None:
```
Comment (rupdeb) [inline code/models/OutputDataModel.py:15]: shouldn't we be following snake casing?

File: code/utils/io/FileWriter.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+class FileWriter:
+    def __init__(self):
+        """
+        File Writer init
+        """
+        LOG.debug(f'{__name__} init')
+        self.outputCoreMetricsDf = None
+        self.outputReadabilityMetricsDf = None
+
+    def __generateCoreMetricsDf(self, coreMetrics: CoreMetrics) -> pd.DataFrame :
```
Comment (rupdeb) [inline code/utils/io/FileWriter.py:21]: would suggest to have all the generate metrics method in a separate class

File: code/models/OutputDataModel.py (hunk 1 above)
Comment (ats0stv) [reply code/models/OutputDataModel.py:15]: Discussed and agreed to go ahead with camelCase

**Comment by chhabriv** [comment]
> why are there so many changes?
Initially had included changes from Arun's PR. 
That should be solved now.

File: code/utils/io/FileWriter.py (hunk 2 above)
Comment (chhabriv) [reply code/utils/io/FileWriter.py]: I agree. Just want to check on where to place it.
Can we merge this and then refactor? 
Will be adding tests as well before closure of the issue.

Comment (rupdeb) [reply code/utils/io/FileWriter.py]: yes, definitely

**Pull request merged**